### ISSUE TITLE
deselectAll() triggers onSelect

### ISSUE THREAD

**Issue by kevin-krug-sociomantic** [sociomantic-tsunami/flounder#110]
## current

by specification :

`onSelect - (function) Triggered when an option selectbox is closed`
(https://github.com/sociomantic-tsunami/flounder/blob/master/README.md#available-config-options)

currently : 

onSelect is also triggered by calling the api method `deselectAll()`. 

## wanted

either the docs could be aligned with the behaviour or fix `deselectAll()` to not trigger `onSelect` according to current docs.

Imo, i'm in favour of api methods such as `deselectAll()` not triggering handlers like on `onSelect`. It would be good to be sure that the handlers are triggered only for "real" ui actions and not by programmatic api method calls, analog to:

`setByValue( value, multiple ) ...This does not fire the onClick event.`

console test on the demo page http://flounderjs.com/demo/ :

```
var _f = $('.flounder')[0].flounder
undefined
_f
Flounder {props: Object, originalTarget: input#vanilla--input--tags.vanilla--input--tags.flounder--hidden, target: div, allowHTML: false, flounderClass: ""…}
_f.onSelect = function(){console.log('yo onSelect was called')}
(){console.log('yo onSelect was called')}
VM2985:1 yo onSelect was called // ui selection
_f.deselectAll()
VM2985:1 yo onSelect was called
undefined
_f.setByValue(["All", "Tags" ], true );
[Array[1], null]
```

**Comment by mousemke** [comment]
that is actually the difference between `setByValue` and `clickByValue`.  should this be a new function for language purposes? it would be very easy to pass true or not for programmatic

**Comment by kevin-krug-sociomantic** [comment]
@mousemke currently as per docs one would assume that `deselectAll()` doesn't fire clicks/`onSelect`. is there a reason for `deselectAll()` to trigger `onSelect`? in any case it would be great to have `deselectAll()` to not trigger clicks with or without passing another param, with or without a name change.

**Comment by kevin-krug-sociomantic** [comment]
@mousemke in recent work with flounder i stumbled over this again. imo, i still would expect programmatic api methods such as `deselectAll()` and `setByValue` should not fire change events triggering handlers. what do you reckon? will this be changed? or if not, will there be an option to pass for a programmatic change?

**Comment by mousemke** [comment]
This is a UX issue and there's no straight answer.  Though if it's been tripping you up, it sounds like it may need to be changed.  I'll run the question past some other folk and come back with an answer in a day of so.

**Comment by kevin-krug-sociomantic** [comment]
@mousemke :boom: thanks for the quick support!

**Comment by mousemke** [comment]
ok @kevin-krug-sociomantic .  loose plan

`onSelect` is going to get a name change for clarity to `onChange`.  `onSelect` would give a deprication warning, but would not be removed until the next major version.

`deselectAll` will continue to fire the `onChange` event, because by the new language it's changing, but I will give it a `silent` bool you can pass, so the doc would be `deselectAll( [ silent ] )`

does that sound adequate?

**Comment by kevin-krug-sociomantic** [comment]
@mousemke thanks! 💐 works for me!

**Comment by mousemke** [comment]
closed in `1.1.0`
